tests/unit/rules/javascript/test.no_eval.js

```diff
@@ -52,7 +52,10 @@ describe('no_eval', () => {
     },
     {
       code: '(function(exe){ exe("foo") })(eval);',
-      message: ['eval can be harmful.'],
+      // The code snippet above will be parsed as script and we'll get the
+      // parsing as module error in the message. We use
+      // `expect.stringContaining()` to ignore the error message part.
+      message: [expect.stringContaining('eval can be harmful.')],
```

Comment:
Can't you be a little bit more precise to make sure that all messages are as expected?

Reply:
I was thinking more about writing tests for the parsing error part (the new code) once we agree on the implementation.

Reply:
@Rob--W I'm not sure that is really necessary in this particular test case, the change required in this PR seems to be mainly needed to don't assume as part of the assertion that the only thing that the message will contain is the no-eval one (but the core of the assertion is still intact: "we expect the no-eval rule to have added the expected validation message".

Anyway I don't think we need to block on this particular detail, this PR is still a draft and if needed we can discuss more about these details in the actual review passes.